src/functions.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Interop\Http\Middleware;
+
+use ReflectionException;
+use ReflectionFunctionAbstract;
+use TRex\Reflection\CallableReflection;
+
+if (! function_exists('Interop\Http\Middleware\is_delegate')) {
```

Comment:
This is not necessary. The only time we needed to use `function_exists` before defining functions is when namespaces were not available.